src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -265,9 +265,10 @@ private void ValidateParameters(byte[] buffer, int offset, int count)
                     }
 
                     readBytes = await FillBufferAsync(adapter, SecureChannel.ReadHeaderSize + payloadBytes).ConfigureAwait(false);
-                    if (readBytes < 0)
+                    Debug.Assert(readBytes >= 0);
+                    if (readBytes == 0)
```

Comment:
Is it worth continuing to throw in the `< 0` case as before, just for 2.1 out of extra caution ? 
I realize that should be impossible and the read above does not do such a check, but it's conceivable existing code is achieving something?

Reply:
Right I can't see how it could either. I don't feel strongly.